marklogic-data-hub/src/test/java/com/marklogic/hub/flow/VerifyBatchTimestampTest.java

```diff
@@ -0,0 +1,51 @@
+package com.marklogic.hub.flow;
+
+import com.marklogic.client.io.Format;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.step.RunStepResponse;
+import com.marklogic.hub.test.Customer;
+import com.marklogic.hub.test.ReferenceModelProject;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+
+public class VerifyBatchTimestampTest extends AbstractHubCoreTest {
+
+    @Test
+    void test() {
+        installProjectInFolder("test-projects/custom-step-with-sleep");
+        
+        ReferenceModelProject project = new ReferenceModelProject(getHubClient());
+        doWithWriteBatcher(getHubClient().getStagingClient(), writeBatcher -> {
+        	writeBatcher.add(project.buildCustomerInstanceToWrite(new Customer(0, "Jane"), Format.JSON, null));
+        });
+        
+        RunFlowResponse response = runFlow(new FlowInputs("customFlowWithSleep"));
+        RunStepResponse stepResponse = response.getStepResponses().get("1");
+
+        assertEquals("completed step 1", stepResponse.getStatus());
+        assertEquals(1, stepResponse.getSuccessfulBatches());
+        
+        DateTimeFormatter formatter = new DateTimeFormatterBuilder()
```

Comment:
Looks good - and in a future PR, I'll refactor this a bit so we have a helper method to make this easier in future tests.

Reply:
Sounds great.  I'm reopening the PR